src/aof.c

```diff
@@ -116,18 +116,18 @@ aofInfo *aofInfoDup(aofInfo *orig) {
 
 /* Format aofInfo as a string and it will be a line in the manifest. */
 sds aofInfoFormat(sds buf, aofInfo *ai) {
-    if (includeSpace(ai->file_name)) {
-        /* If file_name contains spaces we wrap it in quotes. */
-        return sdscatprintf(buf, "%s \"%s\" %s %lld %s %c\n", 
-            AOF_MANIFEST_KEY_FILE_NAME, ai->file_name, 
-            AOF_MANIFEST_KEY_FILE_SEQ, ai->file_seq, 
-            AOF_MANIFEST_KEY_FILE_TYPE, ai->file_type);
-    } else {
-        return sdscatprintf(buf, "%s %s %s %lld %s %c\n", 
-            AOF_MANIFEST_KEY_FILE_NAME, ai->file_name, 
-            AOF_MANIFEST_KEY_FILE_SEQ, ai->file_seq, 
-            AOF_MANIFEST_KEY_FILE_TYPE, ai->file_type);
-    }
+    sds filename_repr = NULL;
+
+    if (sdsneedsrepr(ai->file_name))
```

Comment:
I fail to see how it'll decide to add quotes for string with spaces. 
As far as I can tell `sdsneedsrepr` will return 0 for such strings. 
Maybe the reason the test passes is that you added newlines to the test. 

I guess the solution is to also check `includeSpace` here. 
Or for better performance (not scan the string twice), add an argument to `sdsneedsrepr` indicating if you want to include spaces or not.

Reply:
Thanks! messed this up indeed. I think of this as a helper for `sdssplitargs()`, so I think it's reasonable to assume spaces always count (like tabs, btw).